lib/Conf.ml

```diff
@@ -1590,12 +1571,10 @@ let parse_state_attr attr =
   | Ok ("disable", _) -> Some `Disable
   | _ -> None
 
-let is_jane_street_local_annotation config name ~test =
-  String.equal test ("extension." ^ name)
-  ||
-  if config.opr_opts.rewrite_old_style_jane_street_local_annotations.v then
-    String.equal test name || String.equal test ("ocaml." ^ name)
-  else false
+let is_jane_street_local_annotation _config name ~test =
```

Comment:
Remove the `_config` argument. Propagating this change to callsites will unlock lots of simplification and should revert most of #38.

I will wait to review until after that change.

(I wasn't sure that this function belongs in `Conf`, but I don't actually see a better spot. Maybe `Syntax`? But that file is so small; it seems like the wrong place, too. So I say keep it here. But if you find a better spot, go for it.)